### ISSUE TITLE
Unable to delete recipe

### ISSUE THREAD

**Issue by stevetasticsteve** [mealie-recipes/mealie#924]
### First Check

- [X] This is not a feature request
- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Mealie documentation, with the integrated search.
- [X] I already read the docs and didn't find an answer.

### What is the issue you are experiencing?

I tried importing this recipe: https://cookwith.co.uk/recipes/heinz-beanz-pulled-pork
The import adds an item to the database with the title [

The entry causes a 404 error. 
The entry cannot be deleted either ("unable to delete recipe"). 
I reproduced the same error in make docker-dev

I'd say I myself am intermediate skill level with django, never used fast api but I'm willing to learn. I'm looking to contribute, would this be a good place to start or am is it out of my depth?

### Deployment

Docker (Linux)

### Deployment Details

Sqlite using docker-compose

### Mealie Version

0.5.3

**Comment by cristiangauma** [comment]
I'm facing the same issue on the same version.

The recipe directory is properly deleted, but the WebUI shows me an error and sometimes it is deleted and sometimes it is not.

Nevertheless, I'm not able to add the same recipe with the same slug again, it just shows me an error. Also, using the API to delete the recipe or to create a new one with the same slug shows me an error.

**Comment by hay-kot** [comment]
Can either of you provide the logs?

**Comment by stevetasticsteve** [comment]
[mealie.log](https://github.com/hay-kot/mealie/files/7852671/mealie.log)

![recipe](https://user-images.githubusercontent.com/38364294/149086315-7097ec57-779e-410f-bbeb-c841de4024f2.png)

Yes, no problem.
I'm unable to create a recipe with the name "[" normally. I'm guessing there's validation on the recipe name, but somehow the import managed to bypass the validation and create a database entry that causes problems.

**Comment by cristiangauma** [comment]
In my case, I'm mostly in the same boat as the faulty recipes also have "[]".

I'm using the API to create recipes that contain '[foo] - bar', but the behavior of Mealie is weird about it. What I do is basically, create the recipe and if it is already created I delete it and then re-create it again. 

My experience with this is that Mealie does what I want, does it without problem but at some point it stops to work, and it happens randomly and not always with the same recipe. 

Doesn't seem to make so much sense, but I could add/update a bunch of recipes without any problem (like, 25-30) and then at some point when I try to delete one of the recipes to upload it again, I cannot delete them anymore, and then I cannot delete **any** of the recipes containing "[]" when I could before. If I recreate the Mealie instance, and start again, it doesn't happen at the same number of recipes imported or with the same recipe.

During another Mealie instance recreation, I even couldn't see the recipe in the WebUI after trying to delete it and giving me an error, but the API returned that it could not be created again because the slug already existed.

TL;DR  what it is failing for me is not the creation itself (I can create recipes with '[]' without any problem), but at some random point in time I cannot delete them anymore.

**Comment by hay-kot** [comment]
> I'd say I myself am intermediate skill level with django, never used fast api but I'm willing to learn. I'm looking to contribute, would this be a good place to start or am is it out of my depth?

Now that I understand what's going on, I think this should be a relatively simple fix, you may have to touch a lot of parts of the code-base though. 

The issue is that during the slugify process mealie converts the recipe name to a url-safe name so "My Recipe" -> "my-recipe". I think all we'd need to do is tie in a validator (maybe in SQLAlchemy?) to validate that the name and slug are both valid. In this case, the slug turns into a blank string so you cannot remove it. 

Something to also consider is that maybe CRUD operations should have a slug OR id based rest endpoints to at least have an escape hatch when something like this happens. 

Let me know if you're interested in taking this one, thanks!

**Comment by stevetasticsteve** [comment]
@hay-kot I am interested, but I don't think I can get to it right away. If someone wants to jump in they should go right ahead. 

I tried to get going with it but couldn't set up the environment. I need to familiarise myself with poetry and node for the frontend. I'm sure I could copy paste instructions to get started but I'd rather have a little time to figure it out and get an overview.
